cacti/test_cacti.py

```diff
@@ -0,0 +1,126 @@
+# (C) Datadog, Inc. 2010-2016
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+# stdlib
+import logging
+import os
+import shutil
+from nose.plugins.attrib import attr
+
+# project
+from shared.test.common import AgentCheckTest, Fixtures
+
+log = logging.getLogger()
+
+CONFIG = """
+init_config:
+
+instances:
+    -   mysql_host: localhost
+        mysql_user: root
+        rrd_path:   /tmp/cacti_test/rrds
+        rrd_whitelist: %s
+""" % Fixtures.file('whitelist.txt')
+
+@attr(requires='cacti')
+class TestCacti(AgentCheckTest):
+
+    CHECK_NAME = 'cacti'
+
+    CACTI_CONFIG = {
+        'mysql_host': 'localhost',
+        'mysql_user': 'root',
+        'rrd_path':   '/tmp/cacti_test/rrds',
+        'rrd_whitelist': "%s" % Fixtures.file('whitelist.txt')
+    }
+
+    def setUp(self):
+        self.tmp_dir = '/tmp/cacti_test'
+        self.rrd_dir = os.path.join(os.path.dirname(__file__), "cacti")
+
+        # Create our temporary RRD path, if needed
+        try:
+            os.mkdir(self.tmp_dir)
+        except Exception:
+            # Ignore, directory already exists
+            pass
+
+    def tearDown(self):
+        # Clean up the temp directory
+        shutil.rmtree(self.tmp_dir)
+
+    def _copy_rrds(self, xml_dir):
+        if os.access("/usr/bin/rrdtool", os.R_OK | os.X_OK):
+            # Copy the latest RRDs from /var/lib/rra/ to the test location
+            shutil.copytree("/var/lib/cacti/rra/", os.path.join(self.tmp_dir, 'rrds'))
+            return True
+        else:
+            return False
+
+    def testChecks(self):
+        config = {
+            'instances': [self.CACTI_CONFIG]
+        }
+
+        check = self.load_check(config)
+        rrd_dir = os.path.join(self.tmp_dir, 'rrds')
+
+        # Restore the RRDs from the XML dumps
+        if not self._copy_rrds(self.rrd_dir):
+            return
```

Comment:
I'm not 100% sure, but I'm afraid we might be exiting here... We might want to add an `assert False` here, to know for sure if that's the case.